### PR TITLE
Refactor 'sided' command creation

### PR DESCRIPTION
**Pull request by aschokking**
* Created a 'Base' command for each subsystem to capture common code like having the subsystem as a class variable and calling requires
* Updated ShooterBelt and ShooterWheel commands to be consistent with Agitator where the command will be passed in it's subsystem in the constructor instead of having the setter method. The reason for this change was many times we had forgotten when setting up commands to call the setter and this way it will be enforced at compile time. 
* If the commands end up needing more and more dependencies to the point where this approach becomes a problem we can update things to use a factory to create commands to insulate callers from any injectable dependencies.

### CODE REVIEW

File: Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java [hunk 1]
```diff
@@ -4,20 +4,18 @@
...
-        this.beltSubsystem = beltSubsystem;
-        this.requires(beltSubsystem);
-        targetBeltSpeed = beltSubsystem.getTargetSpeed();
+    protected final ShooterWheelSubsystem shooterWheelSubsystem;
+    private double targetBeltSpeed;
+
+    public RunBeltIfWheelAtSpeedCommand(ShooterBeltSubsystem shooterBeltSubsystem, ShooterWheelSubsystem shooterWheelSubsystem) {
+        super(shooterBeltSubsystem);
+        this.shooterWheelSubsystem = shooterWheelSubsystem;
+        targetBeltSpeed = shooterBeltSubsystem.getTargetSpeed();
```
Comment (JohnGilb) [inline Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java:18]: This just gets the most recent set target speed, which could be anything (e.g. 0). This needs to read directly from some new DoubleProperty on the BeltSubsystem like `BeltIntakeSpeed`

Comment (JohnGilb) [reply Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java:18]: And you don't need it here, you could just read that in the `execute()`.

File: Competition2017/src/competition/subsystems/shooter_belt/commands/RunShooterBeltCommand.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    
+    @Override
+    public void initialize() {
+        log.info("Initializing " +  this.getName());
+    }
+
+    @Override
+    public void execute() {
+        shooterBeltSubsystem.setTargetSpeed(targetBeltSpeed);
+        shooterBeltSubsystem.updatePeriodicData();
```
Comment (JohnGilb) [inline Competition2017/src/competition/subsystems/shooter_belt/commands/RunShooterBeltCommand.java]: We can delete this now that the shooterBelt is a PeriodicDataSource

File: Competition2017/src/competition/subsystems/shooter_wheel/commands/RunShooterCommand.java [hunk 3]
```diff
@@ -33,8 +23,8 @@ public void initialize(){
     
     @Override 
     public void execute(){
-        shooterWheel.setTargetSpeed(commandedLauncherSpeed.get());
-        shooterWheel.updatePeriodicData();
+        shooterWheelSubsystem.setTargetSpeed(commandedLauncherSpeed.get());
+        shooterWheelSubsystem.updatePeriodicData();
```
Comment (JohnGilb) [inline Competition2017/src/competition/subsystems/shooter_wheel/commands/RunShooterCommand.java]: as above, remove this

**Review by JohnGilb — APPROVED** [review]

File: Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java (hunk 1 above)
Comment (aschokking) [reply Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java:18]: https://app.asana.com/0/38541457243752/270337401932322 should fix this, agreed it's bad

**Pull request merged**